shell/src/main/java/tachyon/shell/TfsShell.java

```diff
@@ -174,6 +178,61 @@ public void load(TachyonURI filePath) throws IOException {
   }
 
   /**
+   * Persist a file or directory currently stored only in Tachyon to the UnderFileSystem
+   *
+   * @param filePath the TachyonURI path to persist to the UnderFileSystem
+   * @throws IOException
+   */
+  public void persist(TachyonURI filePath) throws IOException {
+    try {
+      TachyonFile fd = mTfs.open(filePath);
+      FileInfo fInfo = mTfs.getInfo(fd);
+      if (fInfo.isFolder) {
+        List<FileInfo> files = mTfs.listStatus(fd);
+        List<String> errorMessages = new ArrayList<String>();
+        for (FileInfo file : files) {
+          TachyonURI newPath = new TachyonURI(file.getPath());
+          try {
+            persist(newPath);
```

Comment:
It looks like we call `persist` recursively, which is not entirely true as here `newPath` should be a file. So why don't we have a separate helper function `persistInternal` which only persists a file? In that way, the code looks less complicated, and it might be also easier to unit test.